### PR TITLE
[fixed] OverlayTrigger hover triggers on mousenter/leave

### PR DESCRIPTION
**Pull request by jquense**
Addresses an inadvertent revert we did switching the hover triggers to use mouse{enter|leave} instead of out/over, since those events also fire on child elements, causing flicker. Unfortunately the React mousenter/leave implentation is [broken](https://github.com/facebook/react/issues/4251), so we need to simulate it ourselves.

### CODE REVIEW

File: src/OverlayTrigger.js [hunk 1]
```diff
@@ -195,8 +200,8 @@ const OverlayTrigger = React.createClass({
         '[react-bootstrap] Specifying only the `"hover"` trigger limits the visibilty of the overlay to just mouse users. ' +
         'Consider also including the `"focus"` trigger so that touch and keyboard only users can see the overlay as well.');
 
-      props.onMouseOver = createChainedFunction(this.handleDelayedShow, this.props.onMouseOver, triggerProps.onMouseOver);
-      props.onMouseOut = createChainedFunction(this.handleDelayedHide, this.props.onMouseOut, triggerProps.onMouseOut);
+      props.onMouseOver = createChainedFunction(this.handleDelayedShow, this.handleMouseOver, triggerProps.onMouseOver);
```
Comment (taion) [inline src/OverlayTrigger.js]: Looks like this replaced the wrong callback? i.e. `this.props.onMouseOver` instead of `this.handleDelayedShow`.

Comment (jquense) [inline src/OverlayTrigger.js]: yup, ooops, thanks

**Comment by jquense** [comment]
fixed!

**Comment by taion** [comment]
LGTM. Mind adding a comment explaining this though?

**Comment by jquense** [comment]
good idea

**Pull request merged**